### DISCUSSION TITLE
No users when starting the client

### DISCUSSION THREAD

**Discussion by alequint8912** [ltsp/ltsp#400] [Q&A] [answered]
Hello. My problem is that when creating the image from a virtual machine, the process eliminates my users and the configuration already established. So when starting the client, only an administrator user appears, not the users that I already create, I also lose the configuration

When starting the process, the console shows me this message

 Cleaning up LM18xcfe-flat.vmdk before mksquashfs ...

I create the image with this command

sudo ltsp image LM18xcfe-flat.vmdk

Thanks

**Comment by alkisg** [comment]
LTSP is using SSH and SSHFS to authenticate the users against the server, and to mount their home directories.
So you should create your users on the server, not in the VM image.

If you really really know what you're doing and you don't want LTSP to clean up your VM users, run `man ltsp image` and read about the --cleanup=0 parameter.

**Reply by alequint8912** [reply]
Hi, thanks for the quick reply.

I followed your advice using the parameter --cleanup = 0, it no longer shows the cleanup message, but it still looks the same. I'm pretty new to this topic and with linux, I followed the ltsp.org guide using a virtual machine with virtualbox. Virtualize with OS Linux Mint 18 xcfe 32 bits and run the command sudo ltsp image LM18xcfe-flat.vmdk, getting the same result. I show virtualization image and the resulting image

![image](https://user-images.githubusercontent.com/18752175/107783315-1ee68a80-6d18-11eb-81d4-84e9536a63c4.png)

**Comment by alkisg** [comment]
Hi, create new user accounts on the server, NOT in the image, and run `sudo ltsp initrd` to have them show up in the LTSP client login screen.

**Reply by alequint8912** [reply]
Hello, create the new user on the server and manage to enter the client. The problem now is that I lose all the configuration that I already created in that account in my virtual machine, proxy, desktop background, everything, I also do the configuration in the server account and the client does not receive it either. There is some way to preserve this setting. My server is mounted on Ubunut 20.04 and my Virtual Machine on Linux Mint 18

**Comment by alkisg** [comment]
Hi, login to the booted LTSP client and do any changes that you want.
They should be preserved after client reboot.

Do not bother with any settings inside the VM, these are not used at all.

**Reply by alequint8912** [reply]
Hi, 
I tried some changes in the client, but when restarting it does not save it. I don't know if it requires an additional command to achieve this

**Comment by alkisg** [comment]
No commands are needed at all, it should be automatic.
Log in to a client, type the following commands, and paste all the output here.

```shell
cat /proc/cmdline
mount | grep home
id
ltsp info
```

You may also come to IRC for live support if you want; if so, go to ltsp.org and click on the IRC link.

**Reply by alequint8912** [reply]
This is the output

**cat /proc/cmdline**
initrd=ltsp.img initrd=initrd.img root=/dev/nfs nfsroot=192.168.70.54:/srv/ltsp ltsp.image=images/LM18xcfe-flat.vmdk.img loop.max_part=9 BOOTIF=01-00-22-4d-a0-fb-d5 

mount | grep home
nothing to show

**id**
uid=1000(administrador) gid=1000(administrador) groups=1000(administrador),4(adm),24(cdrom),27(sudo),30(dip),46(plugdev),113(lpadmin),130(sambashare),131(lxd),133(epoptes)

**ltsp info**
LTSP version: 21.01-1~ubuntu20.04.1

SERVER XSESSIONS: xfce.desktop

SERVER OS: PRETTY_NAME="Linux Mint 18"

CHROOTS:


VMs:


IMAGES:

**Comment by alkisg** [comment]
OK now try the following on the server:.

```shell
sudo -i
adduser testuser
...
ltsp initrd
```

Then reboot and login as the testuser. DO NOT add the testuser to the VM; do not even open the VM. After login, run again the `mount | grep home` command and see if it now lists an SSHFS or NFS mount.

I suspect that you used --cleanup=0 after all, and you managed to create a local user, with an empty home, while you want server users, not VM users.

**Reply by alequint8912** [reply]
I did what you asked and the command does not show anything, so I recreated the image without the --cleanup = 0 parameter. I login with the user testuser, and the command still doesn't show anything

**Comment by alkisg** [comment]
Boot the client. Login as administrator. Then open a terminal and type `su - testuser`, and enter his password.
Does that su command produce any errors or warnings?

**Reply by alequint8912** [reply]
no error, login successful

**Comment by alkisg** [comment]
@alequint8912, I would suggest that you come to IRC to check the issue with remote support.
If you want to, go to ltsp.org and click on the IRC link.

**Reply by alequint8912** [reply]
Ok thanks for the help

**Comment by alkisg** [comment, marked as answer]
The problem was that Linux Mint is using --no-install-recommends, breaking Debian policy, and this resulted in SSHFS not being installed.

**Reply by alequint8912** [reply]
Thanks you very much!!!
